### ISSUE TITLE
Error 500 on gif upload

### ISSUE THREAD

**Issue by xakan** [mastodon/mastodon#21330]
### Steps to reproduce the problem

1. Upgrade your Simple Hosting instance to v4.0.2 (on a Debian 11 host up to date)
2. Try to upload a gif
3. See the error

### Expected behaviour

The gif should be uploaded

### Actual behaviour

An error 500 is displayed

### Detailed description

After upgrading to Mastodon v4.0.2, users reported that they were unable to upload a gif, even if it respects the restrictions.

First I thought that Tusky was in cause because it wasn't updated to support this new version, but I also reproduce from a browser with another gif.

I inspected mastodon-sidekiq output from journalctl and nothing appeared. I only found it for mastodon-web:
```
Nov 21 15:56:34 CTmastodon bundle[592394]: [8e9143f0-3ddb-48a7-9d5f-0ecf1707b158] [paperclip] Trying to link /tmp/RackMultipart20221121-592394-9pxuhg.gif to /tmp/d70dcb16ec03e82ad9911a7f31cf2d9720221121-592394-wsry7n.gif
Nov 21 15:56:34 CTmastodon bundle[592394]: [8e9143f0-3ddb-48a7-9d5f-0ecf1707b158] [paperclip] Trying to link /tmp/d70dcb16ec03e82ad9911a7f31cf2d9720221121-592394-wsry7n.gif to /tmp/390a4c8d8f1ad7e54bbc68b0e819862c20221121-592394-70oinw.gif
Nov 21 15:56:34 CTmastodon bundle[592394]: [8e9143f0-3ddb-48a7-9d5f-0ecf1707b158] Command :: file -b --mime '/tmp/390a4c8d8f1ad7e54bbc68b0e819862c20221121-592394-70oinw.gif'
Nov 21 15:56:34 CTmastodon bundle[592394]: [8e9143f0-3ddb-48a7-9d5f-0ecf1707b158] Command :: ffmpeg -nostdin -i '/tmp/d70dcb16ec03e82ad9911a7f31cf2d9720221121-592394-wsry7n.gif' -loglevel 'fatal' -movflags 'faststart' -pix_fmt 'yuv420p' -vf 'scale='\''trunc(iw/2)*2:trunc(ih/2)*2'\''' -vsync 'cfr' -c:v 'h264' -maxrate '1300K' -bufsize '1300K' -frames:v '10800' -crf '18' -map_metadata '-1' -acodec 'aac' -strict 'experimental' -y '/tmp/d70dcb16ec03e82ad9911a7f31cf2d9720221121-592394-wsry7n20221121-592394-knaq7f.mp4'
Nov 21 15:56:34 CTmastodon bundle[592394]: [8e9143f0-3ddb-48a7-9d5f-0ecf1707b158] method=POST path=/api/v2/media format=html controller=Api::V2::MediaController action=create status=500 duration=304.76 view=1.39 db=1.63
```

Didn't find anything in nginx log either, except a simple error 500.

### Specifications

Mastodon v4.0.2
Ruby 3.0.3p157 (because 3.0.4 wasn't available)
PostgreSQL 13.8
Redis 6.0.16
Nodejs v16.18.1

Host :
Debian 11 with stable repository

All browsers and apps are concerned here.

The issue wasn't here before upgrade.

**Comment by bpawel-bclub** [comment]
Can you upload other types of media? I had 500 when permissions on public and public/system were not set correctly.

**Comment by Unkn0wnNode** [comment]
I've been only able to upload images. Media like videos and gif's result in "500 Error processing thumbnail for uploaded media". I am also running 4.0.2.

**Comment by ineffyble** [comment]
What version of ffmpeg do you have installed?

**Comment by xakan** [comment]
@ineffyble the issue is not closed.  I have ffmpeg v7:4.3.5-0 installed on my host, and still encounter the issue.

**Comment by ineffyble** [comment]
Ah, my apologies, thought the above comment was from the original reporter.

**Comment by ineffyble** [comment]
@xakan can you try running one of the ffmpeg commands from the logs on the server yourself and see what the output is?

But change `-loglevel 'fatal'` to `-loglevel 'info'`

**Comment by xakan** [comment]
I don't encounter any issue when I manually run the same command on another .gif file :
```lang=bash
mastodon@CTmastodon:~$ ffmpeg -nostdin -i 'giphy2.gif' -loglevel 'info' -movflags 'faststart' -pix_fmt 'yuv420p' -vf 'scale='\''trunc(iw/2)*2:trunc(ih/2)*2'\''' -vsync 'cfr' -c:v 'h264' -maxrate '1300K' -bufsize '1300K' -frames:v '10800' -crf '18' -map_metadata '-1' -acodec 'aac' -strict 'experimental' -y 'bla.mp4'
ffmpeg version 4.3.5-0+deb11u1 Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 10 (Debian 10.2.1-6)
  configuration: --prefix=/usr --extra-version=0+deb11u1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-pocketsphinx --enable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
Input #0, gif, from 'giphy2.gif':
  Duration: 00:00:03.28, start: 0.000000, bitrate: 2111 kb/s
    Stream #0:0: Video: gif, bgra, 308x246, 12.50 fps, 12.50 tbr, 100 tbn, 100 tbc
Stream mapping:
  Stream #0:0 -> #0:0 (gif (native) -> h264 (libx264))
[libx264 @ 0x55edb82ff000] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2
[libx264 @ 0x55edb82ff000] profile High, level 2.0, 4:2:0, 8-bit
[libx264 @ 0x55edb82ff000] 264 - core 160 r3011 cde9a93 - H.264/MPEG-4 AVC codec - Copyleft 2003-2020 - http://www.videolan.org/x264.html - options: cabac=1 ref=3 deblock=1:0:0 analyse=0x3:0x113 me=hex subme=7 psy=1 psy_rd=1.00:0.00 mixed_ref=1 me_range=16 chroma_me=1 trellis=1 8x8dct=1 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=-2 threads=6 lookahead_threads=1 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=3 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=2 keyint=250 keyint_min=12 scenecut=40 intra_refresh=0 rc_lookahead=40 rc=crf mbtree=1 crf=18.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 vbv_maxrate=1300 vbv_bufsize=1300 crf_max=0.0 nal_hrd=none filler=0 ip_ratio=1.40 aq=1:1.00
Output #0, mp4, to 'bla.mp4':
  Metadata:
    encoder         : Lavf58.45.100
    Stream #0:0: Video: h264 (libx264) (avc1 / 0x31637661), yuv420p, 308x246, q=-1--1, 12.50 fps, 12800 tbn, 12.50 tbc
    Metadata:
      encoder         : Lavc58.91.100 libx264
    Side data:
      cpb: bitrate max/min/avg: 1300000/0/0 buffer size: 1300000 vbv_delay: N/A
[mp4 @ 0x55edb82fde40] Starting second pass: moving the moov atom to the beginning of the file
frame=   41 fps=0.0 q=-1.0 Lsize=     244kB time=00:00:03.04 bitrate= 656.6kbits/s speed=8.34x    
video:242kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.527020%
[libx264 @ 0x55edb82ff000] frame I:1     Avg QP:17.89  size: 14292
[libx264 @ 0x55edb82ff000] frame P:13    Avg QP:17.46  size:  7560
[libx264 @ 0x55edb82ff000] frame B:27    Avg QP:18.63  size:  4995
[libx264 @ 0x55edb82ff000] consecutive B-frames:  7.3%  9.8% 14.6% 68.3%
[libx264 @ 0x55edb82ff000] mb I  I16..4: 12.5% 45.6% 41.9%
[libx264 @ 0x55edb82ff000] mb P  I16..4:  4.4% 17.9% 23.5%  P16..4: 25.0% 14.7%  9.6%  0.0%  0.0%    skip: 4.9%
[libx264 @ 0x55edb82ff000] mb B  I16..4:  1.9%  4.6%  9.1%  B16..8: 34.4% 19.1%  7.9%  direct: 8.4%  skip:14.6%  L0:46.0% L1:41.8% BI:12.2%
[libx264 @ 0x55edb82ff000] 8x8 transform intra:36.1% inter:47.8%
[libx264 @ 0x55edb82ff000] coded y,uvDC,uvAC intra: 72.4% 91.6% 86.6% inter: 40.8% 56.5% 31.0%
[libx264 @ 0x55edb82ff000] i16 v,h,dc,p: 59% 15% 18%  8%
[libx264 @ 0x55edb82ff000] i8 v,h,dc,ddl,ddr,vr,hd,vl,hu: 28% 12% 24%  6%  6%  7%  5%  7%  6%
[libx264 @ 0x55edb82ff000] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 25% 17% 17%  6%  7%  8%  7%  7%  6%
[libx264 @ 0x55edb82ff000] i8c dc,h,v,p: 41% 18% 30% 11%
[libx264 @ 0x55edb82ff000] Weighted P-Frames: Y:7.7% UV:7.7%
[libx264 @ 0x55edb82ff000] ref P L0: 58.7%  7.9% 21.2% 12.2%
[libx264 @ 0x55edb82ff000] ref B L0: 85.4% 12.3%  2.3%
[libx264 @ 0x55edb82ff000] ref B L1: 96.3%  3.7%
[libx264 @ 0x55edb82ff000] kb/s:603.49
mastodon@CTmastodon:~$ ls
bla.mp4  giphy2.gif  live
```

**Comment by ineffyble** [comment]
Are you using local/file storage, or a storage backend like S3?

**Comment by xakan** [comment]
I'm on a local storage, it's a simple kimsufi from ovh, and a lxc container.

**Comment by ClearlyClaire** [comment]
Isn't there anything more in the `mastodon-web` logs? Isn't there a backtrace along the logged error? This would be very helpful to figure out what is going on.

**Comment by xakan** [comment]
Unfortunately, there’s nothing more. Here is what I get with `journalctl -u mastodon-web |grep ffmpeg -B10 -A10` :
```
Nov 21 16:51:09 CTmastodon bundle[592386]: [2509545f-13e3-494d-b9a0-fb3e5a6ba836] method=GET path=/api/v1/timelines/home format=html controller=Api::V1::Timelines::HomeController action=show status=200 duration=10.30 view=0.51 db=1.72
Nov 21 16:51:09 CTmastodon bundle[592386]: [b47322a7-a006-4a01-a750-339a0bd35988] method=GET path=/api/v1/notifications format=html controller=Api::V1::NotificationsController action=index status=200 duration=13.65 view=0.42 db=3.77
Nov 21 16:51:09 CTmastodon bundle[592394]: [45251761-e476-4080-bc23-175040b82fc6] method=GET path=/api/v1/announcements format=html controller=Api::V1::AnnouncementsController action=index status=200 duration=8.73 view=1.19 db=2.80
Nov 21 16:51:09 CTmastodon bundle[592394]: [2688056f-f9f6-400a-ae60-d33374cd9a0d] method=POST path=/api/v1/markers format=html controller=Api::V1::MarkersController action=create status=200 duration=27.49 view=0.30 db=16.37
Nov 21 16:51:11 CTmastodon bundle[592386]: [3fdb88ce-0db8-4610-9268-820507b85121] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=12.13 view=0.00 db=1.33 key=https://mastodon.gougere.fr/users/R1Rail#main-key
Nov 21 16:51:12 CTmastodon bundle[592386]: [41c5eb9b-23a1-4d3f-8d65-ca6ab085db3a] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=18.15 view=0.00 db=2.32 key=https://mastodon.social/users/Looping#main-key
Nov 21 16:51:12 CTmastodon bundle[592386]: [b8f7e322-ce25-4d30-84bb-105d71843fed] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=14.52 view=0.00 db=1.50 key=https://mastodon.social/users/Looping#main-key
Nov 21 16:51:13 CTmastodon bundle[592394]: [812e09f6-8dda-4e2b-8095-6d9fb7cba708] [paperclip] Trying to link /tmp/RackMultipart20221121-592394-lddbhu.gif to /tmp/d70dcb16ec03e82ad9911a7f31cf2d9720221121-592394-qwn8ur.gif
Nov 21 16:51:13 CTmastodon bundle[592394]: [812e09f6-8dda-4e2b-8095-6d9fb7cba708] [paperclip] Trying to link /tmp/d70dcb16ec03e82ad9911a7f31cf2d9720221121-592394-qwn8ur.gif to /tmp/ccb0e394fd251724ac333472ce91136f20221121-592394-iw3v5x.gif
Nov 21 16:51:13 CTmastodon bundle[592394]: [812e09f6-8dda-4e2b-8095-6d9fb7cba708] Command :: file -b --mime '/tmp/ccb0e394fd251724ac333472ce91136f20221121-592394-iw3v5x.gif'
Nov 21 16:51:13 CTmastodon bundle[592394]: [812e09f6-8dda-4e2b-8095-6d9fb7cba708] Command :: ffmpeg -nostdin -i '/tmp/d70dcb16ec03e82ad9911a7f31cf2d9720221121-592394-qwn8ur.gif' -loglevel 'fatal' -movflags 'faststart' -pix_fmt 'yuv420p' -vf 'scale='\''trunc(iw/2)*2:trunc(ih/2)*2'\''' -vsync 'cfr' -c:v 'h264' -maxrate '1300K' -bufsize '1300K' -frames:v '10800' -crf '18' -map_metadata '-1' -acodec 'aac' -strict 'experimental' -y '/tmp/d70dcb16ec03e82ad9911a7f31cf2d9720221121-592394-qwn8ur20221121-592394-gux6zl.mp4'
Nov 21 16:51:13 CTmastodon bundle[592394]: [812e09f6-8dda-4e2b-8095-6d9fb7cba708] method=POST path=/api/v2/media format=html controller=Api::V2::MediaController action=create status=500 duration=312.73 view=0.68 db=2.19
Nov 21 16:51:20 CTmastodon bundle[592386]: [4e07a978-8d57-484f-8738-56212a635f2c] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=18.22 view=0.00 db=2.28 key=https://scicomm.xyz/users/SciTania#main-key
Nov 21 16:51:27 CTmastodon bundle[592394]: [222c4ebe-853c-4c55-b417-44101be594f4] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=21.74 view=0.00 db=2.73 key=https://mastodon.social/users/Mediapart#main-key
Nov 21 16:51:30 CTmastodon bundle[592386]: [17b09f44-947f-4f5f-8379-72ba9e339500] method=GET path=/users/xakan/statuses/109377161182546779 format=json controller=StatusesController action=show status=200 duration=644.59 view=15.91 db=10.08 key=https://pixelfed.social/i/actor#main-key
Nov 21 16:51:31 CTmastodon bundle[592386]: [d2367218-66c5-44a4-b54d-1684fb3e2911] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=25.36 view=0.00 db=2.21 key=https://mastodon.social/users/paul_denton#main-key
Nov 21 16:51:31 CTmastodon bundle[592386]: [0748a69b-ad9b-4cb8-9e39-c0bc7781023c] method=GET path=/users/xakan format=json controller=AccountsController action=show status=200 duration=23.59 view=10.13 db=1.93 key=https://pixelfed.social/i/actor#main-key
Nov 21 16:51:32 CTmastodon bundle[592394]: [b66dbe11-d751-4a7d-973e-3863bf8ace07] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=18.24 view=0.00 db=2.12 key=https://imaginair.es/users/nikolavitch#main-key
Nov 21 16:51:33 CTmastodon bundle[592394]: [a261111f-5921-4fb3-af5b-92e1065e2444] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=19.34 view=0.00 db=2.42 key=https://mastodon.social/users/paul_denton#main-key
Nov 21 16:51:34 CTmastodon bundle[592386]: [06433d92-f3bd-446d-acbd-c2d50a81b0a7] method=GET path=/@Lo/with_replies format=*/* controller=AccountsController action=show status=200 duration=42.38 view=25.96 db=6.39
Nov 21 16:51:35 CTmastodon bundle[592386]: [3a566789-729d-474e-b9d1-20b7b810e12a] method=POST path=/inbox format=html controller=ActivityPub::InboxesController action=create status=202 duration=19.00 view=0.00 db=2.15 key=https://oc.todon.fr/users/piou#main-key
```

I don’t notice any error.

**Comment by ClearlyClaire** [comment]
Oh, right, I forgot this was the only place that could return a 500 without an uncaught exception…

Applying the following patch and reloading `mastodon-web` should allow you to get a backtrace that will hopefully be helpful:
```patch
diff --git a/app/controllers/api/v2/media_controller.rb b/app/controllers/api/v2/media_controller.rb
index 288f847f1..e330971bd 100644
--- a/app/controllers/api/v2/media_controller.rb
+++ b/app/controllers/api/v2/media_controller.rb
@@ -6,7 +6,5 @@ class Api::V2::MediaController < Api::V1::MediaController
     render json: @media_attachment, serializer: REST::MediaAttachmentSerializer, status: @media_attachment.not_processed? ? 202 : 200
   rescue Paperclip::Errors::NotIdentifiedByImageMagickError
     render json: file_type_error, status: 422
-  rescue Paperclip::Error
-    render json: processing_error, status: 500
   end
 end
```

**Comment by newdayhost** [comment]
We're also experiencing this after upgrading to 4.0.2 (from 3.5.3)
We applied the patch and here's the backtrace.

GIF we tried to upload: https://media.tenor.com/dUcjIzlxL0IAAAAC/aww-shucks-bashful.gif

```Nov 24 02:14:55 mastodon bundle[1170859]: [f210cc8b-30cf-45b4-b462-4a8465bc245e]
Nov 24 02:14:55 mastodon bundle[1170859]: [f210cc8b-30cf-45b4-b462-4a8465bc245e] Paperclip::Error (Error while transcoding 681cffb890121a27ca5f8eec886e444520221124-1170859-ezgd9y: Command 'ffmpeg -nostdin -i '/tmp/681cffb890121a27ca5f8eec886e444520221124-1170859-ezgd9y.gif' -loglevel 'fatal' -movflags 'faststart' -pix_fmt 'yuv420p' -vf 'scale='\''trunc(iw/2)*2:trunc(ih/2)*2'\''' -vsync 'cfr' -c:v 'h264' -maxrate '1300K' -bufsize '1300K' -frames:v '10800' -crf '18' -map_metadata '-1' -acodec 'aac' -strict 'experimental' -y '/tmp/681cffb890121a27ca5f8eec886e444520221124-1170859-ezgd9y20221124-1170859-ngcc5z.mp4'' returned 159. Expected 0
Nov 24 02:14:55 mastodon bundle[1170859]: Here is the command output: STDOUT:
Nov 24 02:14:55 mastodon bundle[1170859]: STDERR:
Nov 24 02:14:55 mastodon bundle[1170859]: Bad system call
Nov 24 02:14:55 mastodon bundle[1170859]: ):
Nov 24 02:14:55 mastodon bundle[1170859]: [f210cc8b-30cf-45b4-b462-4a8465bc245e]
Nov 24 02:14:55 mastodon bundle[1170859]: [f210cc8b-30cf-45b4-b462-4a8465bc245e] lib/paperclip/transcoder.rb:58:in `rescue in make'
Nov 24 02:14:55 mastodon bundle[1170859]: [f210cc8b-30cf-45b4-b462-4a8465bc245e] lib/paperclip/transcoder.rb:54:in `make'
Nov 24 02:14:55 mastodon bundle[1170859]: [f210cc8b-30cf-45b4-b462-4a8465bc245e] lib/paperclip/gif_transcoder.rb:109:in `make'
Nov 24 02:14:55 mastodon bundle[1170859]: [f210cc8b-30cf-45b4-b462-4a8465bc245e] lib/paperclip/attachment_extensions.rb:21:in `block in post_process_style'
Nov 24 02:14:55 mastodon bundle[1170859]: [f210cc8b-30cf-45b4-b462-4a8465bc245e] lib/paperclip/attachment_extensions.rb:20:in `each'
Nov 24 02:14:55 mastodon bundle[1170859]: [f210cc8b-30cf-45b4-b462-4a8465bc245e] lib/paperclip/attachment_extensions.rb:20:in `inject'
Nov 24 02:14:55 mastodon bundle[1170859]: [f210cc8b-30cf-45b4-b462-4a8465bc245e] lib/paperclip/attachment_extensions.rb:20:in `post_process_style'
Nov 24 02:14:55 mastodon bundle[1170859]: [f210cc8b-30cf-45b4-b462-4a8465bc245e] app/controllers/api/v2/media_controller.rb:5:in `create'
Nov 24 02:14:55 mastodon bundle[1170859]: [f210cc8b-30cf-45b4-b462-4a8465bc245e] app/controllers/concerns/localized.rb:11:in `set_locale'
Nov 24 02:14:55 mastodon bundle[1170859]: [f210cc8b-30cf-45b4-b462-4a8465bc245e] lib/mastodon/rack_middleware.rb:9:in `call'```

**Comment by cleeyv** [comment]
This can be fixed by removing `@resources` from the `SystemCallFilter` line of the mastodon-web.service systemd config file, as was done by https://github.com/mastodon/mastodon/commit/a9ff5c830951eead9c4cfa04532d378220ccd8e4 in [`dist/mastodon-web.service`](https://github.com/mastodon/mastodon/blob/main/dist/mastodon-web.service) more than a year ago. For instances that were installed earlier than that, their old systemd services will result in this error unless manually changed. 

A documentation change that could be made to help avoid this kind of problem in the future is to add something to the upgrade page https://github.com/mastodon/documentation/blob/master/content/en/admin/upgrading.md about updating the systemd service files to reflect the latest versions in https://github.com/mastodon/mastodon/tree/main/dist.

**Comment by xakan** [comment]
Indeed, I replaced the mastodon-*.service files as suggested by @cleeyv , and everything is now working fine.

Assuming it's the solution to this issue, I close it in the same time. Thanks all for your help!

**Comment by xakan** [comment]
Indeed, I replaced the mastodon-*.service files as suggested by @cleeyv , and everything is now working fine.

Assuming it's the solution to this issue, I close it in the same time. Thanks all for your help!

**Comment by ClearlyClaire** [comment]
Yeah, this was caused by a change introduced in 3.4.0 and fixed in 3.5.0, and the issue would trigger more often in 4.0 than 3.5 because we changed `POST /api/v2/media` to synchronously process some media files that were previously processed in background jobs, with a different set of system call restrictions that would not cause the issue.

I'm going to add a note in the 3.5.0 and 4.0.0 release notes accordingly.
